engine/resources/symbolic.go

```diff
@@ -0,0 +1,109 @@
+package resources
+
+import (
+	"fmt"
+	"log"
+	"strings"
+)
+
+// symbolicMode calculates the filemodes from the symbolic file input
+func symbolicMode(p string) (mode string, err error) {
+	var octal int64
+	var out int64
+	m := parse(p)
+	for _, v := range m {
+		v = string(v)
+		switch v[0] {
+		default:
+			err = fmt.Errorf("error")
+			return v, err
+		case 'g':
+			switch v[2] {
+			case 'r':
+				octal = groupRead
+			case 'w':
+				octal = groupWrite
+			case 'x':
+				octal = groupExecute
+			}
+		case 'o':
+			switch v[2] {
+			case 'r':
+				octal = otherRead
+			case 'w':
+				octal = otherWrite
+			case 'x':
+				octal = otherExecute
+			}
+		case 'u':
+			switch v[2] {
+
+			case 'r':
+				octal = userRead
+			case 'w':
+				octal = userWrite
+			case 'x':
+				octal = userExecute
+			}
+		}
+		out = out + octal
+		if err != nil {
+			log.Fatal("Invalid perm")
```

Comment:
You should never fatal deep inside the code. If at all, then only at the top-level. This would cause the whole program to die and potentially corrupt any in progress database operation or write. We always return an error and propagate upwards.